Codeinwp/neve
# Feat/featured post

### Summary
Added basic functionality for the featured post in Neve

### Will affect visual aspect of the product
NO

### Screenshots 
<img width="1680" alt="Screenshot 2022-02-25 at 16 07 18" src="https://user-images.githubusercontent.com/9929553/155729491-0f6c0031-2f7f-4330-9ef5-ff96985c7367.png">
<img width="1680" alt="Screenshot 2022-02-25 at 16 12 02" src="https://user-images.githubusercontent.com/9929553/155729592-491aa779-05e0-4317-b2d1-f8f0acf0e65d.png">



### Test instructions
- Go to Customizer -> Layouts -> Blog / Archive
- Enable featured post
- Test with/without sidebars
- Test with different post layouts ( List and grid will look the same )
- Test with the homepage set as latest post/frontpage
- Test with posts with/without image
- Test with different pagination types

NOTE: This will work without Neve Pro. This PR is based on Bogdan's PR for post-type enhancements but it does not add support for post-type enhancements yet. It should be merged after we assure "post-type enhancements" is merged.


Closes Codeinwp/neve-pro-addon#1105.
